Modix.Bot/ModixBot.cs

```diff
@@ -224,14 +231,23 @@ private async Task HandleCommand(SocketMessage messageParam)
 
             var context = new CommandContext(_client, message);
 
-            using (var scope = _scope.ServiceProvider.CreateScope())
-            {
-                await scope.ServiceProvider
-                    .GetRequiredService<IAuthorizationService>()
-                    .OnAuthenticatedAsync(context.User as IGuildUser);
+            var commandScope = _scope.ServiceProvider.CreateScope();
+            _commandScopes[context] = commandScope;
 
-                var result = await _commands.ExecuteAsync(context, argPos, scope.ServiceProvider);
+            await commandScope.ServiceProvider
+                .GetRequiredService<IAuthorizationService>()
+                .OnAuthenticatedAsync(context.User as IGuildUser);
+
+            await _commands.ExecuteAsync(context, argPos, commandScope.ServiceProvider);
```

Comment:
Should the scope for this context be disposed at some point after this line?

Reply:
It gets disposed after the command finishes, in HandleCommandResultAsync